enterprise/backend/test/metabase_enterprise/sso/integrations/saml_test.clj

```diff
@@ -579,10 +581,33 @@
                        (:RelayState params-map))))
               (testing "\nPOST request should redirect to the original redirect URL"
                 (do-with-some-validators-disabled
-                  (fn []
-                    (let [req-options (saml-post-request-options (saml-test-response)
-                                                                 (:RelayState params-map))
-                          response    (client-full-response :post 302 "/auth/sso" req-options)]
-                      (is (successful-login? response))
-                      (is (= redirect-url
-                             (get-in response [:headers "Location"]))))))))))))))
+                 (fn []
+                   (let [req-options (saml-post-request-options (saml-test-response)
+                                                                (:RelayState params-map))
+                         response    (client-full-response :post 302 "/auth/sso" req-options)]
+                     (is (successful-login? response))
+                     (is (= redirect-url
+                            (get-in response [:headers "Location"])))))))))))
+
+
+      (doseq [redirect-url ["/collection/root"
+                            "/test"
+                            "/"]]
+        (testing (format "\nredirect URL = %s" redirect-url)
+          (mt/with-temporary-setting-values [site-url "http://localhost:3001/path"]
```

Comment:
I think this probably warrants its own top-level test. Otherwise it's not obvious that this is testing the specific case of `site-url` including a subpath. I'd split it out into its own test, give it a descriptive `testing` context string, and stick the issue # in it, like what's done on line 565.

I think you can also test the combination of `redirect-url` being a full URL and `site-url` including a subpath, for comprehensiveness.

Reply:
Makes sense, think `relay-state-e2e-test` should cover `redirect-url` being a full URL.